a3p-integration/proposals/z:acceptance/test-lib/psm-lib.js

```diff
@@ -52,6 +53,72 @@ const PSM_PAIR = NonNullish(process.env.PSM_PAIR).replace('.', '-');
  * @import {Coin} from '@agoric/cosmic-proto/cosmos/base/v1beta1/coin.js';
  */
 
+/**
+ * Given either an array of [string, Value] or
+ * { [labelKey]: string, [valueKey]: Value } entries, or a
+ * Record<string, Value> object, log a concise representation similar to
+ * the latter but hiding implementation details of any embedded remotables.
+ *
+ * @template [Value=unknown]
+ * @param {string} label
+ * @param {Array<[string, Value] | object> | Record<string, Value>} data
+ * @param {(...args: unknown[]) => void} [log]
+ * @returns {void}
+ */
+export const logRecord = (label, data, log = console.log) => {
+  const entries = Array.isArray(data) ? [...data] : Object.entries(data);
+  /** @type {[labelKey: PropertyKey, valueKey: PropertyKey] | undefined} */
+  let shape;
+  for (let i = 0; i < entries.length; i += 1) {
+    let entry = entries[i];
+    if (!Array.isArray(entry)) {
```

Comment:
I'm not confident that I can follow this logic. And I'm not quite sure what it's supposed to do.

I'm looking for examples in ci logs, but I see:
> Integration tests / test-docker-build (pull_request) Skipped

... even though the **force:integration** label is present.

Some examples in the docstring might be nice. Or unit tests.

Reply:
Updated the docstring. Actual output is visible in a prior CI run: https://github.com/Agoric/agoric-sdk/actions/runs/11881729913/job/33106407896#step:10:4506
```
  ✔ swap into IST using agd with default gas (9s)
    ℹ METRICS  {
        anchorPoolBalance: {
          brand: Object @Alleged: BoardRemoteUSDC_axl brand {},
          value: 30010011n,
        },
        feePoolBalance: {
          brand: Object @Alleged: BoardRemoteIST brand {},
          value: 0n,
        },
        mintedPoolBalance: {
          brand: Object @Alleged: BoardRemoteIST brand {},
          value: 30010011n,
        },
        totalAnchorProvided: {
          brand: Object @Alleged: BoardRemoteUSDC_axl brand {},
          value: 0n,
        },
        totalMintedProvided: {
          brand: Object @Alleged: BoardRemoteIST brand {},
          value: 30010011n,
        },
      }
    ℹ BALANCES { [denom]: amount, ... } {
        'ibc/295548A78785A1007F232DE286149A6FF512F180AF5657780FC89C009E2C348F': '300000000',
        ubld: '20000000',
        uist: '250000',
      }
```

Reply:
This is a lot of new code in psm-lib that isn't specific to PSM. I'm not opposed to landing it but please don't spend effort improving these deprecated utils. It would be better spent on `@agoric/client-utils` and getting psm-lib to use that.

incidentally, consider marking `@deprecated` all the functions in psm-lib that aren't about PSM per se.